syne_tune/blackbox_repository/conversion_scripts/scripts/pd1_import.py

```diff
@@ -0,0 +1,315 @@
+            return length * [np.nan]
+        return learning_curve + (length - len(learning_curve)) * [np.nan]
+
+    objectives_evaluations = list()
+    for o in available_objectives:
+        task_data[o] = task_data[o].apply(pad_with_nans, args=(learning_curve_length,))
+        objectives_evaluations.append(np.array(task_data[o].to_list()))
+    objectives_evaluations = np.expand_dims(
+        np.stack(objectives_evaluations, axis=-1), 1
+    )
+
+    fidelity_space = {RESOURCE_ATTR: randint(lower=1, upper=learning_curve_length)}
+
+    return BlackboxTabular(
+        hyperparameters=hyperparameters,
+        configuration_space=CONFIGURATION_SPACE,
+        fidelity_space=fidelity_space,
+        objectives_evaluations=objectives_evaluations,
+        fidelity_values=np.arange(1, learning_curve_length + 1),
+        objectives_names=available_objectives,
+    )
+
+
+class PD1Recipe(BlackboxRecipe):
+    def __init__(self):
+        super(PD1Recipe, self).__init__(
+            name=BLACKBOX_NAME,
+            cite_reference="Pre-trained Gaussian processes for Bayesian optimization. "
+            "Wang, Z. and Dahl G. and Swersky K. and Lee C. and Mariet Z. and Nado Z. and Gilmer J. and Snoek J. and "
+            "Ghahramani Z. 2021.",
+        )
+
+    def _download_data(self):
+        file_name = repository_path / f"{BLACKBOX_NAME}.tar.gz"
+        if not file_name.exists():
+            logger.info(f"Did not find {file_name}. Starting download.")
+            download_file(
+                "http://storage.googleapis.com/gresearch/pint/pd1.tar.gz", file_name
+            )
+        else:
+            logger.info(f"Skip downloading since {file_name} is available locally.")
+
+    def _convert_data(self) -> Dict[str, BlackboxTabular]:
+        with tarfile.open(repository_path / f"{BLACKBOX_NAME}.tar.gz") as f:
+            f.extractall(path=repository_path)
+        data = []
+        for matched in ["matched", "unmatched"]:
+            path = (
+                repository_path
+                / BLACKBOX_NAME
+                / f"pd1_{matched}_phase1_results.jsonl.gz"
+            )
+            with open(path, "rb") as fin:
+                data.append(
+                    pd.read_json(fin, orient="records", lines=True, compression="gzip")
+                )
+        df = pd.concat(data)
+        df["eval_time"] = df["eval_time"].apply(
+            lambda x: None if x is None else np.cumsum(x).tolist()
+        )
+
+        tasks = df[
+            ["dataset", "model", "hps.batch_size", "hps.activation_fn"]
+        ].drop_duplicates()
+        bb_dict = {}
+        for _, task in tasks.iterrows():
+            task_name = "{}_{}{}_batch_size_{}".format(
+                task["dataset"],
+                task["model"],
+                ""
+                if task["hps.activation_fn"] is None
+                else f"_{task['hps.activation_fn']}",
+                task["hps.batch_size"],
+            )
```

Comment:
This one is hard to read, can compute it one line above?

```suggestion
            activation_name = "" if task["hps.activation_fn"] is None else f"_{task['hps.activation_fn']}"
            task_name = "{}_{}{}_batch_size_{}".format(
                task["dataset"],
                task["model"],
                activation_name,
                task["hps.batch_size"],
            )
```